src/hooks/useClick.ts

```diff
@@ -0,0 +1,34 @@
+import { RefObject, useCallback, useEffect } from 'react'
+
+export function useClick(
+  innerRefs: Array<RefObject<HTMLElement>>,
+  innerCallback: (e: MouseEvent) => void,
+  outerCallback: (e: MouseEvent) => void,
+) {
+  const handleClick = useCallback(
+    (e: MouseEvent) => {
+      if (innerRefs.some((target) => isEventIncludedParent(e, target.current))) {
+        innerCallback(e)
+        return
+      }
+      outerCallback(e)
+    },
```

Comment:
imo: ほぼほぼ `useOuterCallback` と同じなので、共通化してもいいように見えました

```suggestion
  outerCallback: (e: MouseEvent) => void,
  innerCallback?: (e: MouseEvent) => void,
) {
  const handleClick = useCallback(
    (e: MouseEvent) => {
      if (innerRefs.some((target) => isEventIncludedParent(e, target.current))) {
        innerCallback && innerCallback(e)
        return
      }
      outerCallback(e)
    },
```

Reply:
@uknmr 当初その対応で行こうとしてたんですが、 MultiComboBoxにも影響してしまうため、一旦バグ修正を優先し、ファイル切り分けました。
いずれ共通化するかもしれません